### PR TITLE
Basic Dataloader

### PR DESCRIPTION
**Pull request by francescofarina**
## Proposed changes

Implementing a very basic Dataloader working with `Dataset` and `Sequence[GraphData]` with basic shuffling options and batching.

There are probably many ways to improve efficiency that we can implement later.

## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

File: mlx_graphs/data/dataloaders.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+from mlx_graphs.data import GraphData, GraphDataBatch, batch
+from mlx_graphs.datasets import Dataset
+
+
+class Dataloader:
+    """
+    Data loader class
+
+    Args:
+        dataset: where to load the data from
```
Comment (TristanBilot) [inline mlx_graphs/data/dataloaders.py:14]:
```suggestion
        dataset: `Dataset` or list of `GraphData` to batch and iterate over
```

File: mlx_graphs/data/dataloaders.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import numpy as np
+
+from mlx_graphs.data import GraphData, GraphDataBatch, batch
+from mlx_graphs.datasets import Dataset
+
+
+class Dataloader:
+    """
+    Data loader class
```
Comment (TristanBilot) [inline mlx_graphs/data/dataloaders.py:11]:
```suggestion
    Default data loader to batch and iterate over multiple graphs.
```

File: mlx_graphs/data/dataloaders.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    Args:
+        dataset: where to load the data from
+        batch_size: how many graphs to load per batch. Defaults to 1
```
Comment (TristanBilot) [inline mlx_graphs/data/dataloaders.py:15]:
```suggestion
        Number of graphs to load per batch. Defaults to 1
```

File: mlx_graphs/data/dataloaders.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        shuffle: whether to reshuffle the data at every epoch. Defaults to False
```
Comment (TristanBilot) [inline mlx_graphs/data/dataloaders.py:16]:
```suggestion
        shuffle: Whether to reshuffle the order of graphs within each batch. Defaults to False
```

**Review by TristanBilot — COMMENTED** [review]
Very nice @francescofarina 🤗 Just added few comments.

**Comment by francescofarina** [comment]
cool, docs updated ;)

**Review by TristanBilot — APPROVED** [review]
Perfect :)

**Pull request merged**